LiverSegments/Resources/UI/LiverSegments.ui

```diff
@@ -99,111 +59,143 @@
-          <property name="baseName">
-           <string>CenterlineSegment</string>
+         </widget>
+        </item>
+        <item row="3" column="4">
+         <widget class="QPushButton" name="showHideButton">
+          <property name="text">
+           <string>Show/Hide</string>
           </property>
-          <property name="noneEnabled">
+          <property name="checkable">
            <bool>true</bool>
           </property>
-          <property name="editEnabled">
+          <property name="checked">
            <bool>true</bool>
           </property>
-          <property name="renameEnabled">
-           <bool>true</bool>
+          <property name="flat">
+           <bool>false</bool>
           </property>
          </widget>
         </item>
-        <item row="2" column="2">
-         <widget class="qSlicerMarkupsPlaceWidget" name="endPointsMarkupsPlaceWidget">
-          <property name="maximumSize">
-           <size>
-            <width>15000</width>
-            <height>40</height>
-           </size>
+        <item row="3" column="1" colspan="3">
+         <widget class="qMRMLSegmentSelectorWidget" name="inputSegmentSelectorWidget">
+          <property name="sizePolicy">
+           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+            <horstretch>0</horstretch>
+            <verstretch>0</verstretch>
+           </sizepolicy>
           </property>
-          <property name="placeMultipleMarkups">
-           <enum>qSlicerMarkupsPlaceWidget::ForcePlaceMultipleMarkups</enum>
+          <property name="segmentationNodeSelectorVisible">
+           <bool>false</bool>
           </property>
          </widget>
         </item>
-        <item row="2" column="3">
+        <item row="5" column="2" colspan="3">
          <widget class="QPushButton" name="addSegmentButton">
           <property name="text">
            <string>Add Vessel Centerline </string>
           </property>
          </widget>
         </item>
-        <item row="1" column="3">
-         <spacer name="horizontalSpacer">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
+        <item row="6" column="2" colspan="3">
+         <widget class="QPushButton" name="calculateVascularTerritoryMapButton">
+          <property name="text">
+           <string>Calculate Vascular Territory Segmentation</string>
           </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
+         </widget>
+        </item>
+        <item row="0" column="0" colspan="2">
+         <widget class="QLabel" name="label_4">
+          <property name="text">
+           <string>Vascular Territory Segmentation:</string>
           </property>
-         </spacer>
+         </widget>
+        </item>
+        <item row="0" column="2" colspan="3">
+         <widget class="ctkComboBox" name="selectedVascularTerritoryMapId">
```

Comment:
This should be a `qMRMLNodeComboBox` (https://apidocs.slicer.org/master/classqMRMLNodeComboBox.html). This would allow the user to create, delete, rename, etc. This will probably imply some minor changes in the logic as you don't have exactly the same methods (e.g., currentIndex is not available).

Reply:
One picky detail is that the collapsible button containing all the sub-widgets shows a little shorter than the rest.